### PR TITLE
Docs: update block spacing details (fixes #8743)

### LINKED ISSUE #8743: `block-spacing` errors on multi line blocks
**Tell us about your environment**

* **ESLint Version:** 4.0.0
* **Node Version:** 8.1.2
* **npm Version:** 4.6.1

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

.eslintrc.js
```js
module.exports = {
    'parserOptions': {
        'ecmaVersion': 2017
    },
    'env': {
        'node': true,
        'es6': true
    },
    'root': true,
    'extends': 'eslint:recommended',
    'rules': {
        'block-spacing': 'error',
		
        //turn off other rules to make output clean
        'max-statements-per-line': 'off',
        'no-unused-vars': 'off'
    }
};
```

**What did you do? Please include the actual source code causing the issue.**
The docs describe the `block-spacing` rule as "Disallow or enforce spaces inside of single line blocks". However, the rule is not limited to single line blocks. It is applied multi line blocks as well.

For example, both functions  below trigger the `block-spacing` rule.

```js
'use strict';
function test() {let i = 0;
    return i;
}

function test2() {//comment
    return 0;
}
```

**What did you expect to happen?**
no errors from `block-spacing` rule

**What actually happened? Please include the actual, raw output from ESLint.**
  2:17  error  Requires a space after '{'  block-spacing
  6:18  error  Requires a space after '{'  block-spacing

✖ 2 problems (2 errors, 0 warnings)
  2 errors, 0 warnings potentially fixable with the `--fix` option.

### PR DESCRIPTION
**Pull request by VictorHom**
[X] Documentation update

**What changes did you make? (Give an overview)**
Change the block spacing doc to explain that it's not specific to single line blocks.

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

File: docs/rules/block-spacing.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
-# Disallow or enforce spaces inside of single line blocks (block-spacing)
+# Disallow or enforce spaces between an open block token and the next token on the same line (block-spacing)
```
Comment (platinumazure) [inline docs/rules/block-spacing.md]: It would be good to mention close block token as well, if that's possible without making this too verbose.

**Comment by not-an-aardvark** [comment]
I don't feel strongly about this, but I feel like the rule description doesn't need to be as verbose as it currently is. I think it would be better to make the description describe what the rule does in very general terms (e.g. "Disallow or enforce spaces inside of blocks"), and then describe the behavior in more detail (if necessary) within the body of the documentation.

File: lib/rules/block-spacing.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ const util = require("../ast-utils");
 module.exports = {
     meta: {
         docs: {
-            description: "enforce consistent spacing inside single-line blocks",
+            description: "enforce consistent spacing on the same line between an open block token and the next token, and between a closing block token and the previous token",
```
Comment (Aladdin-ADD) [inline lib/rules/block-spacing.js]: can it be simpler? just as the title. 😄

Comment (VictorHom) [reply lib/rules/block-spacing.js]: I don't feel strongly about the longer sentence. updated 👍

**Comment by VictorHom** [comment]
@platinumazure when you get a chance, could you give one more review on this pr. happy to revise the documentation as needed :)

**Pull request merged**